Hast.Abstractions/Hast.Vitis.Abstractions/Services/VitisCommunicationService.cs

```diff
@@ -27,9 +31,25 @@ public class VitisCommunicationService : OpenClCommunicationService
             MemoryHandle hostMemoryHandle,
             IHardwareExecutionContext executionContext)
         {
-            var isHbm = data.Length <= 256_000_000 &&
+            var configuration = executionContext
+                .HardwareRepresentation
+                .HardwareGenerationConfiguration
+                .GetOrAddOpenClConfiguration();
+
+            var isHbm = configuration.UseHbm &&
+                data.Length <= HbmSizeBytes &&
                 (executionContext.HardwareRepresentation.DeviceManifest as XilinxDeviceManifest)?.SupportsHbm != false;
-            _logger.LogInformation($"Using HBM: {isHbm}.");
+
+            var implementation = executionContext.HardwareRepresentation.HardwareImplementation;
+            if (isHbm && File.Exists(implementation.BinaryPath + NoHbmFlagExtension))

```

Comment:
A fájlos trükk nincs dokumentálva a Readme-ben.

Reply:
Direkt nem hívtam fel rá a figyelmet, mert implementation detail és elkezdene vele trükközni pedig ez nem arra való mert akkor ilyen felemás helyzet alakulna ki hogy a szoftver más HBM beállítást hisz mint amivel buildelve volt. A usernek arról kell tudnia, hogy a UseHbm configot átállíthatja akkor másféle binary jön létre.

Reply:
Ja OK akkor.